### PR TITLE
Feat/sdk improvements

### PR DESCRIPTION
**Pull request by hugocaillard**
### Description

Three additions:

- add an 'options" argument to `deployContract()` it currently allows to specify the clarity_version, but I made it an options object so that we can add more options in the futures with limited or no breaking changes
- ability to get cost reports and and save it to a json file in tests
- change the vitest-helper with an `initBeforeEach` options controlling when the vm state should be re-initialized

Plus multiple small improvements

Fix: #749

### CODE REVIEW

File: components/clarinet-sdk/src-ts/vitest/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { bool } from "@stacks/transactions/dist/cl";
```
Comment (janniks) [inline components/clarinet-sdk/src-ts/vitest/index.ts:1]: Is this used?

**Review by janniks — APPROVED** [review]
Looking good 😎

File: components/clarinet-sdk/src-ts/vitest/index.ts (hunk 1 above)
Comment (hugocaillard) [reply components/clarinet-sdk/src-ts/vitest/index.ts:1]: Ah, my bad, I removed it but in the follow up PR: https://github.com/hirosystems/clarinet/pull/1171/files#diff-c26bfb8a8ec5511c86cae0ab7e151b28db319eccdc06e2992fb679b4c5a8fb23L1

Marking as resolve for this one

**Comment by hugocaillard** [comment]
Thanks @janniks!

**Pull request merged**